src/Controller/V2/Document/DocumentRoute.ts

```diff
@@ -0,0 +1,223 @@
+
+export class DocumentRoute extends BaseRoute {
+  private documentController = new DocumentController()
+
+  private get basePath(): string {
+    return '/doc'
+  }
+
+  public create(router: Router): void {
+    router.post(
+      `${this.basePath}/:projectID/manuscript/:manuscriptID`,
+      celebrate(createDocumentSchema),
+      AuthStrategy.JsonHeadersValidation,
+      AuthStrategy.JWTAuth,
+      (req: Request, res: Response, next: NextFunction) => {
+        return this.runWithErrorHandling(async () => {
+          await this.createDocument(req, res)
+        }, next)
+      }
+    )
+    router.put(
+      `${this.basePath}/:projectID/manuscript/:manuscriptID`,
+      celebrate(updateDocumentSchema),
+      AuthStrategy.JsonHeadersValidation,
+      AuthStrategy.JWTAuth,
+      (req: Request, res: Response, next: NextFunction) => {
+        return this.runWithErrorHandling(async () => {
+          await this.updateDocument(req, res)
+        }, next)
+      }
+    )
+    router.delete(
+      `${this.basePath}/:projectID/manuscript/:manuscriptID`,
+      celebrate(deleteDocumentSchema),
+      AuthStrategy.JsonHeadersValidation,
+      AuthStrategy.JWTAuth,
+      (req: Request, res: Response, next: NextFunction) => {
+        return this.runWithErrorHandling(async () => {
+          await this.deleteDocument(req, res)
+        }, next)
+      }
+    )
+
+    router.get(
+      `${this.basePath}/:projectID/manuscript/:manuscriptID`,
+      celebrate(getDocumentSchema),
+      AuthStrategy.JsonHeadersValidation,
+      AuthStrategy.JWTAuth,
+      (req: Request, res: Response, next: NextFunction) => {
+        return this.runWithErrorHandling(async () => {
+          await this.getDocument(req, res)
+        }, next)
+      }
+    ),
+      router.post(
+        `${this.basePath}/:projectID/manuscript/:manuscriptID/steps`,
+        celebrate(receiveStepsSchema),
+        AuthStrategy.JsonHeadersValidation,
+        AuthStrategy.JWTAuth,
+        (req: Request, res: Response, next: NextFunction) => {
+          return this.runWithErrorHandling(async () => {
+            await this.receiveSteps(req, res)
+          }, next)
+        }
+      )
+    router.get(
+      `${this.basePath}/:projectID/manuscript/:manuscriptID/listen`,
+      celebrate(listenSchema),
+      AuthStrategy.JsonHeadersValidation,
+      AuthStrategy.JWTAuth,
+      (req: Request, res: Response, next: NextFunction) => {
+        return this.runWithErrorHandling(async () => {
+          await this.listen(req, res)
+        }, next)
+      }
+    )
+    router.get(
+      `${this.basePath}/:projectID/manuscript/:manuscriptID/version/:versionID`,
+      celebrate(getStepsFromVersionSchema),
+      AuthStrategy.JsonHeadersValidation,
+      AuthStrategy.JWTAuth,
+      (req: Request, res: Response, next: NextFunction) => {
+        return this.runWithErrorHandling(async () => {
+          await this.getStepFromVersion(req, res)
+        }, next)
+      }
+    )
+  }
```

Comment:
@mbartenev-atypon & @mnatsheh, do you think its better if we have a collaboration route instead of keeping these in the document route? and we can have the documentID sent in the request body. Or do you prefer we keep them like this?